src/tonic/api/points_internal_api.rs

```diff
@@ -258,7 +258,7 @@ impl PointsInternal for PointsInternalService {
             .ok_or_else(|| Status::invalid_argument("UpdateVectors is missing"))?;
 
         update_vectors(
-            UncheckedTocProvider::new_unchecked(&self.toc),
+            StrictModeCheckedInternalTocProvider::new(&self.toc),
```

Comment:
I'm not super happy about this type name because it is so huge. But I do see that it's not strictly related to this PR.

Let's not bother with it in this PR and keep it like this for now.